### PR TITLE
Add proper lexer for .env files

### PR DESCRIPTION
**Pull request by jaxwilko**
This PR is a bit overkill, but resolves the issue reported in https://github.com/wintercms/laravel-config-writer/issues/1.

The problem was previously each line was parsed as a single entity, i.e. a line could be either a `comment` or a `env` entry, but never two things.

The new approach provided by this PR is to use a proper `Lexer` to convert the source into tokens. By doing this, multiple tokens can exist on the same line.

This ensures that the following:
```env
# WINTERCMS

APP_DEBUG=true
APP_URL="http://localhost"

# HELLO WORLD

APP_KEY="changeme" # Change this

APP_EXAMPLE="test"

#ENV_COMMENT="1"

ENV_TEST="1"

#ENV_COMMENT=1
```
Can be interpreted and updated, preserving the original whitespace and comments.

I've added some additional tests to confirm the newly supported functionality

### CODE REVIEW

File: src/EnvFile.php [hunk 1]
```diff
@@ -12,16 +16,23 @@ class EnvFile implements DataFileInterface
...
-     * @var array<string, int>
+     * @var DataFileLexerInterface
      */
-    protected array $map = [];
+    protected DataFileLexerInterface $lexer;
+
+    /**
+     * Env file printer to convert the ast back to a string
+     *
+     * @var DataFilePrinterInterface
```
Comment (LukeTowers) [inline src/EnvFile.php]:
```suggestion
```

**Review by LukeTowers — CHANGES_REQUESTED** [review]

File: src/EnvFile.php (hunk 1 above)
Comment (jaxwilko) [reply src/EnvFile.php]: It's the style of this repo for PHPStan

File: phpstan.neon [hunk 2]
```diff
@@ -1,5 +1,5 @@
 parameters:
-    level: 6
+    level: 5
```
Comment (jaxwilko) [inline phpstan.neon:3]: @bennothommo how would you feel about this?

Comment (bennothommo) [reply phpstan.neon:3]: Is level 6 where you start having to go into "generics" level detail? If so, I'm happy for you to drop it down.

**Comment by bennothommo** [comment]
@jaxwilko if this is to be a true lexer and parsing an environment file into an AST, I would suggest making the initial environment variable a token, as well as the equals sign (and possibly even the whitespace between), so you get a true representation of each portion of the code.

Another little edge case to consider - I believe you can set an environment variable with no value whatsoever, which is the same as setting the environment variable as an empty string, but it otherwise exists, so it is *not null*.

Example:

```
EMPTY_VAR=
```

**Comment by bennothommo** [comment]
@jaxwilko I've added some additional test cases of valid environment variables. If they're too hard, just revert :P

File: phpstan.neon (hunk 2 above)
Comment (LukeTowers) [reply phpstan.neon:3]: Can we keep it at 6 but ignore the specific rule that is causing issues or does it not really work like that?

**Comment by jaxwilko** [comment]
@bennothommo I thought I would have found time for this earlier, but at least it's done now. I've refactored the lexer to do 2 things differently:

- Accept a string containing the whole source, parsing it token by token instead of the previous way of scanning line by line as `file()` returns.
- Read ENV key and value as separate tokens, for ENV value tokens there's also special handling to allow for multi line variable values

**Review by LukeTowers — APPROVED** [review]

**Comment by LukeTowers** [comment]
LGTM!

**Review by bennothommo — APPROVED** [review]

**Pull request merged**